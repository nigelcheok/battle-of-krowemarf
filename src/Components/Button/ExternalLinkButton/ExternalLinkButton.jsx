import React from 'react';

export function ExternalLinkButton(props) {
  return (
    <a href={props.link} target="_blank">
      <button type="button" className="btn btn-outline-dark font-weight-bold">
        <div className={`fab ${props.icon} mr-2`}/>
        { props.text }
      </button>
    </a>
  );
}
