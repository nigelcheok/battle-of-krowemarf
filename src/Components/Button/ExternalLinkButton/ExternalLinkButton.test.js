import React from 'react';
import { shallow } from 'enzyme';
import { ExternalLinkButton } from './ExternalLinkButton';

describe('ExternalLinkButton', () => {
  it('should render correctly with no props', () => {
    const component = shallow(<ExternalLinkButton/>);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with name and description', () => {
    const component = shallow(<ExternalLinkButton link="http://google.com" icon="test" text="test" description="test description"/>);
    expect(component).toMatchSnapshot();
  });
});
