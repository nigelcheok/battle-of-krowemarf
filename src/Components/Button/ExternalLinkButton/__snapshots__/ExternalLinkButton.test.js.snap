// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLinkButton should render correctly with name and description 1`] = `
<a
  href="http://google.com"
  target="_blank"
>
  <button
    className="btn btn-outline-dark font-weight-bold"
    type="button"
  >
    <div
      className="fab test mr-2"
    />
    test
  </button>
</a>
`;

exports[`ExternalLinkButton should render correctly with no props 1`] = `
<a
  target="_blank"
>
  <button
    className="btn btn-outline-dark font-weight-bold"
    type="button"
  >
    <div
      className="fab undefined mr-2"
    />
  </button>
</a>
`;
